erpnext/education/doctype/course_schedule/course_schedule_calendar.js

```diff
@@ -1,11 +1,12 @@
 frappe.views.calendar["Course Schedule"] = {
 	field_map: {
 		// from_datetime and to_datetime don't exist as docfields but are used in onload
-		"start": "from_datetime",
-		"end": "to_datetime",
+		"start": "from_time",
+		"end": "to_time",
 		"id": "name",
 		"title": "course",
-		"allDay": "allDay"
+		"allDay": "allDay",
+		"schedule_date": "schedule_date"
```

Comment:
```suggestion
		"allDay": "allDay"
```

this bit can be handled in doctype controller like we discussed. 

Basically if the start field is DateTime and schedule date != start date then it can be updated in validating function. (add comment there as it's handling very specific case - calendar updates)